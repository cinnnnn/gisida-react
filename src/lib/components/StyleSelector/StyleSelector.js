import React, { Component } from 'react';
import { connect } from 'react-redux'
import PropTypes from 'prop-types';
import { Actions } from 'gisida';
import './StyleSelector.scss';


const mapStateToProps = (state, ownProps) => {
  
  return {
    styles: state.STYLES,
    showFilterPanel: state.MAP.showFilterPanel
  }
}

export class StyleSelector extends Component {

  changeStyle = (e) => {
    const style = e.target.value;
    this.props.dispatch(Actions.changeStyle(style));
  }

  render() {
    const styles = this.props.styles;
    return (
      <div className="leaflet-left leaflet-top leaflet-right layer-selector" style={{ right: this.props.showFilterPanel ? '250px' : '0'}}>
        <div aria-haspopup="true" className="leaflet-control leaflet-control-layers">
          <a title="styles" className="leaflet-control-layers-toggle"> </a>
          <form className="leaflet-control-layers-list">
            <div className="leaflet-control-layers-base">
              {(styles && styles.length > 0) ?
                styles.map(s =>
                  (<label key={`label_${s.label}`} htmlFor="styles">
                    <input  
                      readOnly
                      key={`input_${s.label}`}
                      type="radio"
                      name="leaflet-base-layers"
                      className="leaflet-control-layers-selector"
                      value={s.url}
                      onClick={e => this.changeStyle(e)}
                      checked={!!s.current}
                    />
                    <span>{s.label}</span>
                  </label>)) : <span>No available styles</span>
              }
            </div>
            <div className="leaflet-control-layers-overlays" />
          </form>
        </div>
      </div>)
  }
}

StyleSelector.propTypes = {
  styles: PropTypes.arrayOf(PropTypes.any).isRequired,
  showFilterPanel: PropTypes.bool
};

export default connect(mapStateToProps)(StyleSelector);
