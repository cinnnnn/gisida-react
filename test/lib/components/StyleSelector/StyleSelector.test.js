import React from 'react';
import { StyleSelector } from '../../../../src/lib/components/StyleSelector/StyleSelector'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const styles = [
  {
    label: 'Satelitte',
    url: 'mapbox://styles/mapbox/satellite-v9',
  },
  {
    label: 'Satelitte Streets',
    url: 'mapbox://styles/mapbox/satellite-streets-v9',
  },
]
const componentWrapper = shallow(
  <StyleSelector styles={styles}/>
);

describe('StyleSelector', () => {
  it('StyleSelector component renderes correctly', () => {
    const json = toJson(componentWrapper)
    expect(json).toMatchSnapshot();
  });
});
