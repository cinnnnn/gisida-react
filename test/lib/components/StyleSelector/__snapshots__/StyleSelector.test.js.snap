// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleSelector StyleSelector component renderes correctly 1`] = `
<div
  className="leaflet-left leaflet-top leaflet-right layer-selector"
  style={
    Object {
      "right": "0",
    }
  }
>
  <div
    aria-haspopup="true"
    className="leaflet-control leaflet-control-layers"
  >
    <a
      className="leaflet-control-layers-toggle"
      title="styles"
    >
       
    </a>
    <form
      className="leaflet-control-layers-list"
    >
      <div
        className="leaflet-control-layers-base"
      >
        <label
          htmlFor="Satelitte"
          key="label_Satelitte"
        >
          <input
            checked={false}
            className="leaflet-control-layers-selector"
            id="Satelitte"
            key="input_Satelitte"
            name="leaflet-base-layers"
            onClick={[Function]}
            readOnly={true}
            type="radio"
            value="mapbox://styles/mapbox/satellite-v9"
          />
          <span>
            Satelitte
          </span>
        </label>
        <label
          htmlFor="Satelitte Streets"
          key="label_Satelitte Streets"
        >
          <input
            checked={false}
            className="leaflet-control-layers-selector"
            id="Satelitte Streets"
            key="input_Satelitte Streets"
            name="leaflet-base-layers"
            onClick={[Function]}
            readOnly={true}
            type="radio"
            value="mapbox://styles/mapbox/satellite-streets-v9"
          />
          <span>
            Satelitte Streets
          </span>
        </label>
      </div>
      <div
        className="leaflet-control-layers-overlays"
      />
    </form>
  </div>
</div>
`;
